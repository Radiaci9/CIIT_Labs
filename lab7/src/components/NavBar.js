import React from 'react'
import { Row, Col, Button } from 'reactstrap'
import PropTypes from 'prop-types'

export default class NavBar extends React.Component {
  static propTypes = {
    items: PropTypes.array.isRequired
  }
  createNavBarItem = (label, click, i) =>
    <Button size='lg' key={'button' + i} name={'button' + i} color='primary' style={{marginRight: '20px'}}
        onClick={e => click()}>{label}</Button>
  render() {
    return (
        <Row className='col-sm-11' style={{display: 'flex', margin: '20px auto 20px auto'}}>
            <Col>
                {
                  this.props.items.map((el, i) => this.createNavBarItem(el.label, el.click, i))
                }
            </Col>
        </Row>
      )
    }
  }