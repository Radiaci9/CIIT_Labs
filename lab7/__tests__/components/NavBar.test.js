import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavBar from '../../src/components/NavBar'

describe('<NavBar />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<NavBar items={[ {label: 'label', click: () => {} } ]}/>)
      const component = wrapper.dive()

      expect(toJson(component)).toMatchSnapshot()
    })
  })

  describe('onClick()', () => {
    test('successfully calls the onClick handler', () => {
      const mockOnClick = jest.fn()
      const wrapper = shallow(
        <NavBar items={[ {label: 'label', click: () => mockOnClick() } ]}/>
      );
      const component = wrapper.dive()
      component.find('[name="button0"]').simulate('click');

      expect(mockOnClick.mock.calls.length).toEqual(1);
    })
  })
})
