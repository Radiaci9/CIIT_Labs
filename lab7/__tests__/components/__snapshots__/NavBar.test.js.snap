// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> render() renders the component 1`] = `
<div
  className="col-sm-11 row"
  style={
    Object {
      "display": "flex",
      "margin": "20px auto 20px auto",
    }
  }
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Button
      color="primary"
      key="button0"
      name="button0"
      onClick={[Function]}
      size="lg"
      style={
        Object {
          "marginRight": "20px",
        }
      }
      tag="button"
    >
      label
    </Button>
  </Col>
</div>
`;
